src/lightning_app/utilities/app_logs.py

```diff
@@ -0,0 +1,118 @@
+import json
+import queue
+import sys
+from dataclasses import dataclass
+from datetime import datetime
+from json import JSONDecodeError
+from threading import Thread
+from typing import Iterator, List, Optional, Tuple
+
+import dateutil.parser
+from websocket import WebSocketApp
+
+from lightning_app.utilities.logs_socket_api import _LightningLogsSocketAPI
+from lightning_app.utilities.network import LightningClient
+
+
+@dataclass
+class _LogEventLabels:
+    app: str
+    container: str
+    filename: str
+    job: str
+    namespace: str
+    node_name: str
+    pod: str
+    stream: Optional[str] = None
+
+
+@dataclass
+class _LogEvent:
+    message: str
+    timestamp: datetime
+    labels: _LogEventLabels
+
+
+def _push_logevents_to_read_queue_callback(component_name: str, read_queue: queue.PriorityQueue):
+    """Pushes _LogEvents from websocket to read_queue.
+
+    Returns callback function used with `on_message_callback` of websocket.WebSocketApp.
+    """
+
+    def callback(ws_app: WebSocketApp, msg: str):
+        # We strongly trust that the contract on API will hold atm :D
+        event_dict = json.loads(msg)
```

Comment:
nit: could you add an example of how the log message json looks like ?

Reply:
I think that it's implicitly stated in the `_LogEvent` class. Including example directly might be also considered implementation leakage, so not sure if we want to include that explicitly here ;) 
```
{'message': '', 'timestamp': '', 'labels': {'app': '', 'container': '', 'filename': '', 'job': '', 'namespace': '', 'node_name': '', 'pod': '', 'stream': ''}}
```